porter/main.py

```diff
@@ -142,10 +145,23 @@ def _initialize_endpoints(eth_endpoint: str, polygon_endpoint: str):
         ):
             BlockchainInterfaceFactory.initialize_interface(endpoint=polygon_endpoint)
 
-    def get_ursulas(self,
-                    quantity: int,
-                    exclude_ursulas: Optional[Sequence[ChecksumAddress]] = None,
-                    include_ursulas: Optional[Sequence[ChecksumAddress]] = None) -> List[UrsulaInfo]:
+    def get_ursulas(
+        self,
+        quantity: int,
+        exclude_ursulas: Optional[Sequence[ChecksumAddress]] = None,
+        include_ursulas: Optional[Sequence[ChecksumAddress]] = None,
+        timeout: Optional[int] = None,
+    ) -> List[UrsulaInfo]:
+        if timeout and timeout > self.MAX_GET_URSULAS_TIMEOUT:
+            self.log.warn(
+                f"Provided sampling timeout ({timeout}s) exceeds "
+                f"maximum ({self.MAX_GET_URSULAS_TIMEOUT}s); "
+                f"using {self.MAX_GET_URSULAS_TIMEOUT}s instead"
+            )
+            timeout = self.MAX_GET_URSULAS_TIMEOUT
+        else:
+            timeout = timeout or self.MAX_GET_URSULAS_TIMEOUT
```

Comment:
Not a blocker, but something we can address in the future: if I understand well, we are not checking in the tests that if the provided timeout > MAX_GET_URSULAS_TIMEOUT, the timeout will be MAX_GET_URSULAS_TIMEOUT.

Reply:
You're right. I'll look into testing that separately. https://github.com/nucypher/nucypher-porter/issues/56